app/services/open_graph.rb

```diff
@@ -9,9 +9,13 @@ class OpenGraph
   attr_reader :page, :tags
 
   DEFAULT_METHODS = %i[description title url].freeze
+  CACHE_EXPIRY_IN_MINUTES = 15
 
   def initialize(url)
-    @page = MetaInspector.new(url)
+    # get the html from cache or fetch new
+    html = fetch_html(url)
+    # MetaInspector can read html
+    @page = MetaInspector.new(url, document: html)
```

Comment:
I'm wondering if the two comments are necessary here, as you've done a great job naming the method and instance variable.

Reply:
I'm with Ridhwana on this.  The comments echo what the code does, you could ✂️ these inline comments unless there's some deeper "Why am I doing this" you want to convey.